agent/recorder.go

```diff
@@ -114,29 +127,24 @@ func (r *SpanRecorder) loop() error {
 				r.logger.Printf("error sending spans: %v\n", err)
 			}
 			ticker.Stop()
-			return nil
+			return err
 		}
 	}
 }
 
 // Sends the spans in the buffer to Scope
 func (r *SpanRecorder) SendSpans() (error, bool) {
-	r.Lock()
-	spans := r.spans
-	r.spans = nil
-	r.Unlock()
-
-	r.totalSend = r.totalSend + 1
+	r.spanSenderMutex.Lock()
+	defer r.spanSenderMutex.Unlock()
```

Comment:
Yes I added this lock because we had a `Flush` func calling SendSpans too, but I ended removing that func, so this mutex is not needed anymore, removing it.

If we are going to call the SendSpans only internally, I will changed it to private.